lib/ecmascript.ts

```diff
@@ -1130,73 +1158,94 @@ function MergeLargestUnitOption<T extends Temporal.DateTimeUnit>(
   return { ...options, largestUnit };
 }
 
-export function ToPartialRecord<B extends AnyTemporalLikeType>(bagParam: B, fieldsParam: ReadonlyArray<keyof B>) {
-  // External callers are limited to specific types, but this function's
-  // implementation uses generic property types. The casts below (and at the
-  // end) convert to/from generic records.
-  const bag = bagParam as Record<PrimitivePropertyNames & keyof B, string | number | undefined>;
-  const fields = fieldsParam as ReadonlyArray<keyof B & PrimitivePropertyNames>;
+type FieldCompleteness = 'complete' | 'partial';
+type AllKeys<U> = Keys<U extends unknown ? U : never>;
+type AnyTemporalKey = Exclude<AllKeys<AnyTemporalLikeType>, symbol>;
+type Keys<T> = T extends object ? keyof T : never;
```

Comment:
Usually `object` is avoided because `null` is `object`. I think `Record<string, unknown>` is most typically used.

Reply:
Great point - done.